src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,110 @@
+package duke;
+
+import duke.commands.Command;
+import duke.exceptions.IncorrectCommandFormatException;
+import duke.exceptions.InvalidIndexException;
+import duke.exceptions.InvalidTimeFormatException;
+import duke.exceptions.MissingDescriptionException;
+import duke.exceptions.UnknownCommandException;
+import duke.parser.CommandParser;
+import duke.parser.DeadlineCommandParser;
+import duke.parser.EventCommandParser;
+import duke.parser.IndexCommandParser;
+import duke.parser.SimpleCommandParser;
+import duke.parser.TodoCommandParser;
+import duke.parser.WordCommandParser;
+
+/**
+ * Represents a parser that parses user input into a Command object.
+ */
+public class Parser {
+
+    /**
+     * Represents the different types of commands and their corresponding parsers.
+     */
+    public enum Commands {
+
+        // @formatter:off
+        /*
+         * The different types of commands and their corresponding parsers.
+         */
+        LIST("list", new SimpleCommandParser("list")),
+        MARK("mark", new IndexCommandParser("mark")),
+        UNMARK("unmark", new IndexCommandParser("unmark")),
+        DELETE("delete", new IndexCommandParser("delete")),
+        FIND("find", new WordCommandParser("find")),
+        TODO("todo", new TodoCommandParser()),
+        DEADLINE("deadline", new DeadlineCommandParser()),
+        EVENT("event", new EventCommandParser());
+        // @formatter:on
+
+        /**
+         * The string representation of the command.
+         */
+        public final String commandString;
+        private final CommandParser commandParser;
+
+        /**
+         * Constructs a new command with the specified string representation and one
+         * parameter.
+         *
+         * @param commandString the string representation of the command
+         * @param pattern the pattern that the command takes
+         */
+        private Commands(String commandString, CommandParser commandParser) {
+            this.commandString = commandString;
+            this.commandParser = commandParser;
+        }
+
+        /**
+         * Returns the command with the specified string representation.
+         * Idea was taken from https://www.baeldung.com/java-enum-values#locating-values
+         *
+         * @param commandString the string representation of the command
+         * @return the command with the specified string representation
+         */
+        public static Commands valueOfLabel(String commandString) {
```

Comment:
Maybe this can be named using a verb?

Reply:
Thanks for the suggestion! I will change this